utils/persistent_mode/persistent_demo_new.c

```diff
@@ -31,17 +31,8 @@
 
 /* this lets the source compile without afl-clang-fast/lto */
 #ifndef __AFL_FUZZ_TESTCASE_LEN
-
-ssize_t       fuzz_len;
-unsigned char fuzz_buf[1024000];
-
-  #define __AFL_FUZZ_TESTCASE_LEN fuzz_len
-  #define __AFL_FUZZ_TESTCASE_BUF fuzz_buf
-  #define __AFL_FUZZ_INIT() void sync(void);
-  #define __AFL_LOOP(x) \
-    ((fuzz_len = read(0, fuzz_buf, sizeof(fuzz_buf))) > 0 ? 1 : 0)
-  #define __AFL_INIT() sync()
-
+  #define AFL_COMPAT
+  #include "persistent_replay.h"
```

Comment:
I used the demo to test and showcase the  self-contained replay functionality, I'm guessing that it was not the right place to do it, and it's better to keep it as it was originally.

I think it would make sense to either:
  + move the self-contained _replay_ demo into a new `persistent_demo_replay.c`,
  + add back the original defines and wrap the two blocks into a conditional block like `#ifdef AFL_PERSISTENT_REPLAY`

Although not directly, I hope this answers your question. 😛

Reply:
yes just put that into persistent_demo_replay.c please :)